Create AWX docsite with RST content

oraNod opened the pull request:
##### SUMMARY
This PR adds AWX documentation in RST source migrated from the private `ansible/product-docs` repository.

This is currently a work in progress. The following items must be completed before final review/merge:

- [x] Modify all images and screenshots to remove downstream references.
- [x] Update all variables to replace downstream product names with community equivalents.
- [x] Remove content related to downstream subscriptions (https://github.com/ansible/product-docs/issues/1862).

##### ISSUE TYPE
- New or Enhanced Feature

##### COMPONENT NAME
 - Docs


##### AWX VERSION
N/A


##### ADDITIONAL INFORMATION
N/A

oraNod committed:
Create AWX docsite with RST content

oraNod committed:
drop import license

oraNod committed:
rm docs.ansible sitemap

oraNod committed:
drop backup and restore

oraNod committed:
drop common import license

oraNod committed:
drop common user data tracking

oraNod committed:
drop common feature support

oraNod committed:
drop admin usability data collection

oraNod committed:
rm license files and missing references

oraNod committed:
rm aap jira ticket numbers

oraNod committed:
rm subscription details

oraNod commented on .gitignore:
```diff
@@ -1,5 +1,6 @@
 # Ignore generated schema
 swagger.json
+!docs/docsite/rst/rest_api/_swagger/swagger.json
```
@AlanCoding @relrod @thedoubl3j Hey, not sure who to ask but thought I might try my luck.

I've included `swagger.json` under the new rst directory but we probably want to generate it as part of the docs build. I thought maybe we could call the Makefile from tox to generate the schema before doing a sphinx-build. 

It might be easier to just try and show it rather than explain. But if you'd have opinions or ideas on an approach I'd really appreciate hearing. Cheers!

thedoubl3j replied:
i am on board with that, especially if you think that is the path to go. I don't have much of an opinion either way but want it to be easy/as straight forward as we can make it. I know some folks had some thoughts around changing the makefile, @TheRealHaoLiu this might be a good opportunity to prevent creep in the makefile/split it out per directory. what are your thoughts?

oraNod replied:
Thanks @thedoubl3j I've been working through a bunch of changes to the content with @tvo318 it's going well but is going to result in a pretty chonky initial PR.

I'm going to remove the `swagger.json` file from this PR and remove this line from the gitignore file. At least that way we won't add anything we'd need to delete later. Once we get stuff merged we can figure out how to generate the schema with tox or whatever without getting in TVo's way.

oraNod committed:
drop RH references in admin

oraNod committed:
drop RH references in quickstart

oraNod committed:
rm RH references in relnotes

oraNod committed:
rm RH references in api ref

oraNod committed:
rm RH references in upgrade

oraNod committed:
rm RH references in userguide

oraNod committed:
fix warns for labels and codeblock formats

oraNod committed:
mv config tower to awx

oraNod committed:
improve sphinx build

tvo318 committed:
Admin changes up to ent_auth.rst and related images

tvo318 committed:
Replaced grapjhic

tvo318 committed:
Cleaned up files and graphics thru ldap_auth

tvo318 committed:
Cleanup through mgmt jobs.

tvo318 committed:
Applied content from PR #1869 and #1871.

oraNod committed:
remove variables no longer used

oraNod committed:
remove unused attributes

AlanCoding commented:
looks like the api checks failed with this:

```
linters: commands[2]> yamllint -s .
./docs/docsite/rst/common/containergroup-sa.yml
  14:1      error    wrong indentation: expected 2 but found 0  (indentation)
  30:1      error    wrong indentation: expected 2 but found 0  (indentation)
```

Would it be too much to patch that indentation up in the file? If not, it can be added to the yamllint exclude list.

oraNod committed:
fix yaml formatting

oraNod commented:
> looks like the api checks failed with this:
> 
> ```
> linters: commands[2]> yamllint -s .
> ./docs/docsite/rst/common/containergroup-sa.yml
>   14:1      error    wrong indentation: expected 2 but found 0  (indentation)
>   30:1      error    wrong indentation: expected 2 but found 0  (indentation)
> ```
> 
> Would it be too much to patch that indentation up in the file? If not, it can be added to the yamllint exclude list.

Thanks! Should be fixed now. It's great to have ci tests catching formatting issues with the content etc.

AlanCoding commented:
Ready to take this out of draft status?

tvo318 committed:
Cleanup through security best practices; with content changes throughout.

tvo318 committed:
The last few files from the admin guide.

tvo318 committed:
Changes for all Quickstart content.

tvo318 committed:
Release notes and upgrade/migration guide cleanup.

tvo318 committed:
Cleanup of all userguide content

tvo318 committed:
Deleted delete button and warning

oraNod committed:
cleaning up some warnings

oraNod committed:
rm generated schema

oraNod committed:
disable swagger in api ref

oraNod committed:
rm ignore exception for docsite swagger.json

oraNod committed:
fix weird rendering

oraNod commented:
> Ready to take this out of draft status?

@AlanCoding Hello. Sorry for the slight delay here, I've been on PTO. @tvo318 has gone through all the content and added several commits and I think we're ready to go. We're looking at a few last relics in the images but should have that sorted out here shortly. Let the reviews commence! Thanks!

tvo318 committed:
Remaining cleanup - tower refs and images, etc

tvo318 committed:
Replaced and renamed hierarchy graphic

tvo318 committed:
Removed hierarchy graphic with tower name.

oraNod committed:
fix img ref

oraNod committed:
drop some red hat images

oraNod committed:
drop unused insights images

oraNod commented:
@tvo318 Thanks for the awesome work to port the content over! I think we're pretty much ready to go. I added a couple of commits to remove a few relic images that were downstream specific. I also dropped some unused insights images. Please take a look. I guess we could prune more stale images but I'd rather do that in a follow up PR. As long as it's appropriate to hand over to the community it should be fine.

One other thing I think we might want to address are some of the lurking instances of "controller" and "tower".

For example, on these lines you can see controller is mentioned. Should these change to AWX?

```
administration/ent_auth.rst:6. Create an optional private key for the controller to use as a service provider (SP) and enter it in the **SAML Service Provider Private Key** field.  
administration/ent_auth.rst:7. Provide the IdP with some details about the controller cluster during the SSO process in the **SAML Service Provider Organization Info** field.
administration/ent_auth.rst:   These fields are required in order to properly configure SAML within the controller
```

Also for the REST API should we `s/<controller-host>/<awx-host>` in places like this?

```
rest_api/authentication.rst:	curl -k -c - https://<controller-host>/api/login/
```

oraNod commented:
@tvo318 One other that I haven't checked yet. Does ``/etc/tower/conf.d/`` change to ``settings/defaults.py``?

tvo318 committed:
Replaced controller with AWX ent_auth.rst

tvo318 committed:
Added ref anchor to api_ref.rst

tvo318 committed:
replaced controller with awx in rest_api/authentication.rst

tvo318 commented:
I also have a few things I saw that maybe we need to still address:
- docs/docsite/_themes/version_dropdown.html (should we remove this since we are just supporting 'latest")?
- docs/docsite/_themes/srtd/layout.html (still has references to AUTOMATION and CONTROLLER)
- docs/docsite/conf.py (pointers to translated versions still have controller in them)
I was told these are the supported AWX versions (but you probably already know that) 
![image](https://github.com/ansible/awx/assets/7514801/5217b1f5-8fcf-4da0-b0d0-77de9cf6ef15)

tvo318 committed:
Removed excess text from configure_awx.rst

Removed all of introductory text and graphic from User Interface section except for the include from the configure_awx.rst file.

tvo318 committed:
Replaced error log section in troubleshooting.rst

Replaced text with new text from operator-docs.

AlanCoding commented:
I tried to build this with `tox -e docs` and I got this result:

```
docs: install_deps> python -I -m pip install -r /home/alancoding/repos/awx/docs/docsite/requirements.txt
.pkg: install_requires> python -I -m pip install 'setuptools>=45' 'setuptools_scm[toml]>=6.2'
.pkg: _optional_hooks> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_sdist> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_wheel> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: install_requires_for_build_wheel> python -I -m pip install wheel
.pkg: prepare_metadata_for_build_wheel> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: build_sdist> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
docs: install_package> python -I -m pip install --force-reinstall --no-deps /home/alancoding/repos/awx/.tox/.tmp/package/1/awx-22.7.1.dev69+g878ce41083.tar.gz
docs: commands[0]> sphinx-build -T -E -W -n --keep-going --color -j auto -c docs/docsite -d docs/docsite/build/doctrees -b html docs/docsite/rst docs/docsite/build/html
Running Sphinx v5.1.1
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 109 source files that are out of date
updating environment: [new config] 109 added, 0 changed, 0 removed
reading sources... [100%] userguide/project-sign .. userguide/workflows                                       
/home/alancoding/repos/awx/docs/docsite/rst/rest_api/authentication.rst:169: ERROR: Unknown target name: "api_start".
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] userguide/organizations .. userguide/workflows                                       
generating indices... genindex done
writing additional pages... search done
copying images... [100%] common/images/workflow-diagram.png                                                   
copying downloadable files... [100%] ../../common/containergroup-sa.yml                                       
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build finished with problems, 1 warning.
docs: exit 1 (6.36 seconds) /home/alancoding/repos/awx> sphinx-build -T -E -W -n --keep-going --color -j auto -c docs/docsite -d docs/docsite/build/doctrees -b html docs/docsite/rst docs/docsite/build/html pid=475519
.pkg: _exit> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
  docs: FAIL code 1 (76.18=setup[69.82]+cmd[6.36] seconds)
  evaluation failed :( (76.32 seconds)
```

I wanted to ask if you could help me figure out that error. It looks like it's at the "api_start" point.

oraNod commented:
> I tried to build this with `tox -e docs` and I got this result:
> 
> ```
> docs: install_deps> python -I -m pip install -r /home/alancoding/repos/awx/docs/docsite/requirements.txt
> .pkg: install_requires> python -I -m pip install 'setuptools>=45' 'setuptools_scm[toml]>=6.2'
> .pkg: _optional_hooks> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
> .pkg: get_requires_for_build_sdist> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
> .pkg: get_requires_for_build_wheel> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
> .pkg: install_requires_for_build_wheel> python -I -m pip install wheel
> .pkg: prepare_metadata_for_build_wheel> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
> .pkg: build_sdist> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
> docs: install_package> python -I -m pip install --force-reinstall --no-deps /home/alancoding/repos/awx/.tox/.tmp/package/1/awx-22.7.1.dev69+g878ce41083.tar.gz
> docs: commands[0]> sphinx-build -T -E -W -n --keep-going --color -j auto -c docs/docsite -d docs/docsite/build/doctrees -b html docs/docsite/rst docs/docsite/build/html
> Running Sphinx v5.1.1
> making output directory... done
> building [mo]: targets for 0 po files that are out of date
> building [html]: targets for 109 source files that are out of date
> updating environment: [new config] 109 added, 0 changed, 0 removed
> reading sources... [100%] userguide/project-sign .. userguide/workflows                                       
> /home/alancoding/repos/awx/docs/docsite/rst/rest_api/authentication.rst:169: ERROR: Unknown target name: "api_start".
> looking for now-outdated files... none found
> pickling environment... done
> checking consistency... done
> preparing documents... done
> writing output... [100%] userguide/organizations .. userguide/workflows                                       
> generating indices... genindex done
> writing additional pages... search done
> copying images... [100%] common/images/workflow-diagram.png                                                   
> copying downloadable files... [100%] ../../common/containergroup-sa.yml                                       
> copying static files... done
> copying extra files... done
> dumping search index in English (code: en)... done
> dumping object inventory... done
> build finished with problems, 1 warning.
> docs: exit 1 (6.36 seconds) /home/alancoding/repos/awx> sphinx-build -T -E -W -n --keep-going --color -j auto -c docs/docsite -d docs/docsite/build/doctrees -b html docs/docsite/rst docs/docsite/build/html pid=475519
> .pkg: _exit> python /home/alancoding/repos/awx/env/lib64/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
>   docs: FAIL code 1 (76.18=setup[69.82]+cmd[6.36] seconds)
>   evaluation failed :( (76.32 seconds)
> ```
> 
> I wanted to ask if you could help me figure out that error. It looks like it's at the "api_start" point.

hey @AlanCoding yes it looks like there's a problem with a Sphinx ref from this commit: https://github.com/ansible/awx/pull/14328/commits/ea46505f0bb833887fc206d835e0596f6df715e6#diff-2a3bdf96e33ef2c6f65306a401b9874e7b7a1acdd7af9ea124deee3b040a26afR170

The fix is to modify that to use the `ref:` role. I'll push a commit.

Thanks for catching this. After we get this merged I'd like to add a step to CI to verify the docs build so we can get PR checks for this stuff too.

oraNod committed:
fix sphinx ref

oraNod committed:
drop the old theme

oraNod committed:
drop old static content

oraNod commented:
> I also have a few things I saw that maybe we need to still address:
> 
>     * docs/docsite/_themes/version_dropdown.html (should we remove this since we are just supporting 'latest")?
> 
>     * docs/docsite/_themes/srtd/layout.html (still has references to AUTOMATION and CONTROLLER)
> 
>     * docs/docsite/conf.py (pointers to translated versions still have controller in them)
>       I was told these are the supported AWX versions (but you probably already know that)
>       ![image](https://user-images.githubusercontent.com/7514801/264088293-5217b1f5-8fcf-4da0-b0d0-77de9cf6ef15.png)

Nice one @tvo318 I've removed the old theme and static content. They're no longer needed with the Ansible community theme! I guess we still need to figure out translations but we can do that when we set things up on Read The Docs. We don't need those translation strings in the `conf.py` anyway as they were pulled in by the old layout.

AlanCoding reviewed (approved):
We understand that there's still some things to iterate on, but everyone is excited about getting this content and the build system in.

tvo318 commented:
Remaining tower references to keep or revisit later (unsure of AWX equivalent):
- **rst/userguide/projects.rst:** (points to example `helloworld.yml`` file available at: https://github.com/ansible/tower-example.git)
- **rst/userguide/inventory_plugins_templates.rst:** (refers to plugin: awx.awx.tower for RHAAP inventory source)
- **rst/userguide/overview.rst:** (refers to plugin: awx.awx.tower inventory plugin)
- **rst/userguide/ee_reference.rst:** (refers to plugin: awx.awx.tower inventory plugin)
- **rst/userguide/instance_groups.rst**: (xrefs a public blog)
- **rst/administration/ent_auth.rst:** (refers to variable name)
- **rst/administration/logging.rst:** (refers to variable name)
- **rst/administration/troubleshooting.rst:** (references a file directory)
- **rst/rest_api/api_ref.rst:** (in the stylesheet line 51)

tvo318 committed:
minor tower removals and replacements.

oraNod commented:
Thanks @tvo318 I forgot that the api reference (api_ref.rst) was pulling in some assets from the static folder but we can restore them - and make sure they're all good for the community docs - when we add in the swagger generation step.

tvo318 commented:
Remaining controller references to keep or revisit later (unsure of AWX equivalent):
- **rst/userguide/best_practices.rst:** (references controllercli guide since not sure what new CLI guide link is)
- **rst/userguide/inventories.rst:** (contained in a URL and a reference to the )
- **rst/userguide/job_templates.rst:** (contained in a URL)
- **rst/administration/ent_auth.rst:** (contained in abbreviation for TACACS+)  
- **rst/administration/tipsandtricks.rst:** (references the CLI guide)

tvo318 committed:
Replaced controller terminology and removed excess text.
